src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageItemControl.xaml.cs

```diff
@@ -27,39 +27,19 @@ public PackageItemControl()
         private void CheckBox_Toggled(object sender, RoutedEventArgs e)
         {
             var itemCheckBox = sender as CheckBox;
-            if (itemCheckBox is null || itemCheckBox.Visibility != Visibility.Visible)
-            {
-                return;
-            }
-
-            var parentContentPresenter = VisualTreeHelper.GetParent(this) as ContentPresenter;
-            if (parentContentPresenter is null)
-            {
-                return;
-            }
-
-            var parentBorder = VisualTreeHelper.GetParent(parentContentPresenter) as Border;
-            if (parentBorder is null)
-            {
-                return;
-            }
-
-            var itemContainer = VisualTreeHelper.GetParent(parentBorder) as ListBoxItem;
+            var itemContainer = itemCheckBox?.FindParent<ListBoxItem>();
```

Comment:
Since I can't find where this method is relevant to the projects list changes in this PR, please revert the file. 
If you have refactoring suggestions unrelated to the Issue, we should have a separate issue tracking and have discussion around that.

Reply:
It's used in src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/SolutionView.xaml.cs line 306 of this iteration.

Reply:
Pulled up https://github.com/NuGet/NuGet.Client/pull/3879/files#diff-16de17305f46c11c521245379e35ac238c259620d6bd1894ede8fa2ca66a8216R306 and I do not see `PackageItemControl` being used there. Maybe you referenced the wrong code?

Reply:
What method were you referring to in your original comment, please? The method call on the line you commented on is "FindParent" which is used in SolutionView.xaml.cs. Am I missing something?

Reply:
I commented on `PackageItemControl` being changed to use `FindParent`. Sorry it was unclear.
I can't see any reason that `PackageItemControl` needs to be modified for this PR.

Reply:
Ah, thanks for the clarification. I refactored it so both this and the SolutionView.xaml.cs would take advantage of FindParent. Please reach out to me offline if you would like to discuss further!